liquibase-couchbase/src/main/java/liquibase/ext/couchbase/precondition/SqlCheckCountPrecondition.java

```diff
@@ -0,0 +1,52 @@
+package liquibase.ext.couchbase.precondition;
+
+import com.couchbase.client.java.json.JsonObject;
+import com.couchbase.client.java.query.QueryResult;
+import liquibase.changelog.DatabaseChangeLog;
+import liquibase.database.Database;
+import liquibase.ext.couchbase.database.CouchbaseConnection;
+import liquibase.ext.couchbase.exception.precondition.SqlCheckCountPreconditionException;
+import liquibase.ext.couchbase.operator.ClusterOperator;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import java.util.List;
+
+/**
+ * A precondition that checks sql query returns expected count. The sql++ query need to be written by select ... as count template in order
```

Comment:
Maybe it would be more readable to wrap select ... as count with [] or smth else to separate code from text